### PR TITLE
Feature 3: Zooming and scrolling

### PR DESCRIPTION
**Pull request by angelala3252**
**Features included:**

- View Menu is shown when a page is displayed: contains Zoom To x% options, Zoom In/Out options, Reset Zoom (back to 100%), and Center Page (back to original starting position)
image
image

- Zoom In/Out uses a list of scale factor options, and jumps to the next smallest/largest zoom option depending on whether we are zooming in or out. There are max and min zoom options, and once these are reached the program will only zoom in/out in the opposite direction. 
- Scrolling vertically and horizontally (hold shift) by the amount scrolled
- Zooming in/out by scrolling while holding control (calls the same method as the menu button Zoom In/Out)

Notes:
- unsure about having all the zooming functionality in Page class as it is already quite chunky even without the zooming. however, I do need to access the mediaLayer attribute of the Page.
- unsure about the validity of the Clean Architecture structure for this as I put my methods directly in the GUI layer since I never have to edit any core data classes and I am only changing the GUI.

Testing will be done in a separate pull request.

### CODE REVIEW

File: src/main/java/gui/page/Page.java [hunk 1]
```diff
@@ -36,7 +35,7 @@ public class Page extends StackPane implements MediaObserver, Zoomable {
     private Map<Long, GUIMedia> contents;
 
     private Bounds prevVisibleBounds;
-    private Scale scale;
+    private final Scale scale;
```
Comment (TammyCarrick) [inline src/main/java/gui/page/Page.java:38]: I believe final variables should be all caps.

File: src/main/java/gui/page/Page.java [hunk 2]
```diff
@@ -331,30 +327,49 @@ public void zoomInOrOut(String inOrOut){
...
                 i--;
             }
             // once the loop ends, we have the index of a factor that is less than the current one
-            // if we zoom out we can then plug this factor right into our zoomToFactor function
         }
         this.zoomToFactor(zoomOptions[i]);
     }
 
-    /** given the x and y coords of a point, make that point the top left point of the visible box
+    // commented out for future use
```
Comment (TammyCarrick) [inline src/main/java/gui/page/Page.java:340]: Nice, this is a good to indicate code that's still needs work; I gotta do this

**Review by TammyCarrick — COMMENTED** [review]
- consider adding comments to some methods to make it more clear (most sections are documented well)
- all caps on final varialbes
- nicely done

**Comment by angelala3252** [comment]
Resolves #6

**Review by LiamOdero — APPROVED** [review]
Looks pretty good for the most part. My only comment would be to only commit files that you actually edited since it makes it easier to see what you actually changed for review purposes

**Pull request merged**